### PR TITLE
CIV-15290 CoSC - set CoSCApplicationStatus to Active on successful payment of application fee

### PR DESCRIPTION
**Pull request by hmcts-version1-ignacio**
### I WANT TO set the CoSCApplicationStatus to Active on successful payment for (FEE0459)
SO THAT I can process the application accordingly ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[X] No
```

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java [hunk 1]
```diff
@@ -341,6 +341,7 @@ public class CaseData implements MappableObject {
     private YesOrNo respondentResponseDeadlineChecked;
     //Case name for manage case
     private String caseNameGaInternal;
+    private YesOrNo coSCApplicationStatus;
```
Comment (sherlynkhaw) [inline src/main/java/uk/gov/hmcts/reform/civil/model/CaseData.java:344]: delete this. not needed

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/GeneralApplicationAfterPaymentCallbackHandler.java [hunk 2]
```diff
@@ -57,8 +60,13 @@ private CallbackResponse generalAppAfterPayment(CallbackParams callbackParams) {
             return getCallbackResponse(caseDataBuilder);
         }
 
-        caseDataBuilder.businessProcess(BusinessProcess
-                                            .ready(INITIATE_GENERAL_APPLICATION_AFTER_PAYMENT));
+        if (featureToggleService.isCoSCEnabled()) {
+            caseDataBuilder.businessProcess(BusinessProcess
+                                                .ready(INITIATE_COSC_APPLICATION_AFTER_PAYMENT));
```
Comment (sherlynkhaw) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/GeneralApplicationAfterPaymentCallbackHandler.java:65]: INITIATE_COSC_APPLICATION_AFTER_PAYMENT
INITIATE_GENERAL_APPLICATION_AFTER_PAYMENT

needs to be in ccd definition

File: src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/GeneralApplicationAfterPaymentCallbackHandler.java [hunk 3]
```diff
@@ -57,8 +60,13 @@ private CallbackResponse generalAppAfterPayment(CallbackParams callbackParams) {
             return getCallbackResponse(caseDataBuilder);
         }
 
-        caseDataBuilder.businessProcess(BusinessProcess
-                                            .ready(INITIATE_GENERAL_APPLICATION_AFTER_PAYMENT));
+        if (featureToggleService.isCoSCEnabled()) {
```
Comment (sherlynkhaw) [inline src/main/java/uk/gov/hmcts/reform/civil/handler/callback/user/GeneralApplicationAfterPaymentCallbackHandler.java:63]: this is wrong. 
how do we differentiate between cosc and genapps?

**Review by LeonardoPalmeiro — DISMISSED** [review]

File: src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/GeneralApplicationAfterPaymentCallbackHandlerTest.java [hunk 4]
```diff
@@ -51,14 +54,26 @@ public class GeneralApplicationAfterPaymentCallbackHandlerTest extends BaseCallb
     private GeneralApplicationAfterPaymentCallbackHandler handler;
     @Autowired
     private ObjectMapper objectMapper;
-
     @MockBean
     private GaForLipService gaForLipService;
+    @MockBean
+    private FeatureToggleService featureToggleService;
```
Comment (kalachandrasekar1) [inline src/test/java/uk/gov/hmcts/reform/civil/handler/callback/user/GeneralApplicationAfterPaymentCallbackHandlerTest.java:60]: featureToggleService is not used

**Review by kalachandrasekar1 — DISMISSED** [review]

**Review by Gareth40343 — DISMISSED** [review]

**Pull request merged**